lib/decidim/monkey_patches/mailer/notification_mailer_patches.rb

```diff
@@ -1,30 +1,35 @@
 # frozen_string_literal: true
 module NotificationMailerPatch
 
-    def send_custom_email(event, event_class_name, resource, user, extra)
-        with_user(user) do
-            @organization = resource.organization
-            event_class = event_class_name.constantize
-            @event_instance = event_class.new(resource: resource, event_name: event, user: user, extra: extra)
-            subject = @event_instance.email_subject
-    
-            mail(to: user.email, subject: subject, template_name: extra[:template])
-        end
+  def event_received(event, event_class_name, resource, user, extra) # To avoid event_received from Core to override all of our methods, we use event_received as default method then redirect to corresponding custom method when it's needed.
+    if extra[:participation_moderated]
```

Comment:
I don't understand why we need an extra method here for this case, the only thing I see is `@justification` which is accessible via extra from the `event` class.
@ludivinecp Could you shed some light here ? 😀

Reply:
We'll need specifics templates for this case. Because you have to deal with 3 cases : refused, accepted, modified. A specific method for this seems much cleaner, also 3 differents templates for each case in order to avoid too many conditions.